src/blocks/errors.ts

```diff
@@ -111,3 +111,37 @@ export class RemoteExecutionError extends BusinessError {
     super(message);
   }
 }
+
+/**
+ * Throwable for providing control flow for the temporary panel.
+ *
+ * Like "CancelError", it is not an error, but needs to subclass Error for our exceptional control flow to work.
+ *
+ * Uses type and detail fields to match the CustomEvent interface.
+ *
+ * @see CustomEvent
+ */
+export class SubmitPanelAction extends CancelError {
+  override name = "SubmitPanelAction";
+
+  /**
+   * A custom action type to resolve the panel with, e.g., "submit" or "cancel".
+   */
+  type: string;
+
+  /**
+   * Extra data to resolve the panel with.
+   */
+  detail: JsonObject;
+
+  /**
+   * Create a throwable action for a temporary panel that resolves the panel.
+   * @param type A custom action type to resolve the panel with, e.g., "submit" or "cancel".
+   * @param detail Extra data to resolve the panel with.
+   */
+  constructor(type: string, detail: JsonObject = {}) {
```

Comment:
You can use TS shorthand here, then explicit property declaration is not needed:
```suggestion
  constructor(public readonly type: string, public readonly detail: JsonObject = {}) {
```

See for details: https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/

Reply:
TIL